src/main.py

```diff
@@ -138,13 +138,41 @@ async def handle_event(event: Dict, event_context, project_id_owner: Optional[st
 
 async def process_project_metrics(context: Context, project_id: str, services: List[GCPService]):
     context.log(project_id, f"Starting processing...")
+    await check_x_goog_user_project_header_permissions(context, project_id)
     ingest_lines = await fetch_ingest_lines_task(context, project_id, services)
     fetch_data_time = time.time() - context.start_processing_timestamp
     context.fetch_gcp_data_execution_time[project_id] = fetch_data_time
     context.log(project_id, f"Finished fetching data in {fetch_data_time}")
     await push_ingest_lines(context, project_id, ingest_lines)
 
 
+async def check_x_goog_user_project_header_permissions(context: Context, project_id: str):
+    if project_id in context.use_x_goog_user_project_header:
+        return
+
+    service_usage_booking = os.environ['SERVICE_USAGE_BOOKING'] if 'SERVICE_USAGE_BOOKING' in os.environ.keys() \

```

Comment:
Please add `SERVICE_USAGE_BOOKING` env variable to table in HACKING.MD. @pawelsiwek do we want to update the README with info about new permission as well?